dev/build_arrow.sh

```diff
@@ -38,15 +36,14 @@ function install_arrow_deps {
 }
 
 function build_arrow_cpp() {
- if [ -n "$1" ]; then
-   BUILD_TYPE=$1
- fi
  pushd $ARROW_PREFIX/cpp
 
  cmake_install \
        -DARROW_PARQUET=ON \
        -DARROW_FILESYSTEM=ON \
        -DARROW_PROTOBUF_USE_SHARED=OFF \
+       -DARROW_DEPENDENCY_USE_SHARED=OFF \
+       -DARROW_DEPENDENCY_SOURCE=BUNDLED \
```

Comment:
> -DARROW_DEPENDENCY_SOURCE=BUNDLED

does this change really necessary? some deps may already installed.

Reply:
@Yohahaha, thanks for your comment. If not bundled, we need some extra code to support resolve those installed dependencies and link them. Considering arrow libs are generally built once then installed, currently we can simply use BUNDLED to get a lib of all arrow's bundled dependencies.

Reply:
> Considering arrow libs are generally built once then installed

it's important, hope we can reduce changes of arrow...

Reply:
@Yohahaha, I will take a look.